src/Compilers/Core/Portable/DiagnosticAnalyzer/IAnalyzerAssemblyResolver.cs

```diff
@@ -2,21 +2,39 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if NET
+
 using System.Reflection;
+using System.Runtime.Loader;
 
 namespace Microsoft.CodeAnalysis
 {
     /// <summary>
-    /// Allows a host to override how assembly resolution is performed by the <see cref="AnalyzerAssemblyLoader"/>.
+    /// This interface allows hosts to control exactly how a given <see cref="AssemblyName"/> is resolved to an
+    /// <see cref="Assembly"/> instance. This is useful for hosts that need to load assemblies in a custom way like
+    /// Razor or stream based loading.
     /// </summary>
     internal interface IAnalyzerAssemblyResolver
```

Comment:
I regret you can't call this interface IAnalyzerAssemblyLoader since it's actually doing the actual loading more or less. (i.e. the implementations all become some variant of ALC.Load*), but the names are already taken. :frowning>